remove ads module

LeifWebber opened the pull request.

LeifWebber committed:
docs: add some project desc.

LeifWebber committed:
Merge remote-tracking branch 'origin/main' into rm-ad
chore(merge): merge branch 'main'

LeifWebber committed:
Merge remote-tracking branch 'origin/main' into rm-ad

LeifWebber committed:
feat: remove ads module

crisp-su commented on packages/app-lib/src/state/dirs.rs:
```diff
@@ -24,7 +24,7 @@ impl DirectoryInfo {
     pub fn get_initial_settings_dir() -> Option<PathBuf> {
         Self::env_path("THESEUS_CONFIG_DIR")
-            .or_else(|| Some(dirs::data_dir()?.join("ModrinthApp")))
+            .or_else(|| Some(dirs::data_dir()?.join("ModrinthAppDev")))
```
你修复这个问题的方式是否有点过于粗暴了（
如果在特定开发环境下真的需要修改这个值，可以使用 `THESEUS_CONFIG_DIR` 环境变量来指定一个目录（作为临时修复方案）。
或者可以加入一个特殊的 `DEV_MODE` 环境变量，并在其值为 `1` 时进行开发环境相关的配置，包括更改存储目录。

再一个这种更改应该属于破坏性变更的，你不仅没有进行标注，还和其它 commit 混一起了（

crisp-su replied:
或者作为另一种方案，你也可以这么做：
```rust
pub fn get_initial_settings_dir() -> Option<PathBuf> {
    Self::env_path("THESEUS_CONFIG_DIR").or_else(|| {
        Some(dirs::data_dir()?.join(if cfg!(debug_assertions) {
            "ModrinthAppDev"
        } else {
            "ModrinthApp"
        }))
    })
}
```
这将使程序在 release 编译目标下使用 `ModrinthApp`，而在 debug 目标下则使用 `ModrinthAppDev`。
判断将在编译阶段进行，不会影响程序的运行逻辑。

LeifWebber replied:
Your proposals are all solid. Whether it's about the `DEV_MODE` solution, using **environment variables**, or the **source-level modifications** you suggested, they all align with common modern development practices. I appreciate your thorough approach.

crisp-su commented on packages/app-lib/theseus.db:
为什么这里面有一个二进制文件？
经过我的测试，在删除此文件后程序仍然可以正常运行。
我也没有找到更多说明，这很有可能是一个错误提交。

LeifWebber replied:
To resolve a database migration issue, I cleaned up a bunch of files I don't even remember.   
Then, I performed a series of operations as follows, which might be the reason this file was created.

```bash
cd packages/app-lib
# absolute path here
env:DATABASE_URL="sqlite:C:/Projects/Trileaf-MC/trileaf-launcher/packages/app-lib/theseus.db"
cargo sqlx database create
cargo sqlx migrate run
cargo sqlx prepare
```

LeifWebber committed:
chore: Optimized the .gitignore by adding two locations that should not be included in the **VCS**.

LeifWebber committed:
Resolved the issue between the development directory and the regular directory by leveraging Tauri's debug_assertions mechanism.

LeifWebber committed:
chore: untrack .sqlx directories and .db files

Remove SQLx runtime data and database files from version control as they should not be included in the repository.
